packages/ant-stack/src/utils/schema-tools.ts

```diff
@@ -0,0 +1,37 @@
+const days = ["Su", "M", "Tu", "W", "Th", "F", "Sa"];
+
+/**
+ * Input to a transform function.
+ */
+export type TransformInput = string | string[] | undefined;
+
+/**
+ * Output of a transform function.
+ */
+export type TransformOutput = string[] | undefined;
+
+/**
+ * Get unique, sorted array of strings.
+ * @param value String of comma-separated values or array of such strings.
+ */
+export const flattenStringsAndSplit = (value: TransformInput): TransformOutput =>
+  value
+    ? [
+        ...new Set(Array.isArray(value) ? value.flatMap((x) => x.split(",")) : value.split(",")),
+      ].sort()
```

Comment:
I think I actually prefer `Array.from` syntax more than array spreading, just because the brackets can sometimes cause syntax confusion (i.e. expressions surrounded by brackets or parentheses immediately followed by more brackets/parentheses can be interpreted as a function call, which Prettier commonly addresses with a semicolon)

```suggestion
    ? Array.from(new Set(Array.isArray(value) ? value.flatMap((x) => x.split(",")) 
    : value.split(","))).sort()
```